src/main/java/com/formulasearchengine/mathmlquerygenerator/BasicXQueryGenerator.java

```diff
@@ -64,10 +74,9 @@ private String generateSimpleConstraints(Node node, boolean isRoot) {
                 //If an element node and not an attribute or text node, add to xquery and increment index
                 childElementIndex++;
 
-                if (handleSpecialElements(child, childElementIndex) || child.getLocalName() != null
-                        && XMLHelper.ANNOTATION_XML_PATTERN.matcher(child.getLocalName()).matches()) {
-                    //Ignore annotations and presentation mathml
-                } else {
+                // Only consider content mathml and ignore annotations or presentation mathml
+                if (!handleSpecialElements(child, childElementIndex) && (child.getLocalName() == null
```

Comment:
this is a complex condition. maybe it would be a good idea to use some boolean temporary variables with semantic reasonable names to make it easier readable